compiler/rustc_typeck/src/astconv/mod.rs

```diff
@@ -2786,6 +2837,11 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
                     .map_or(false, |s| s.trim_end().ends_with('<'));
 
             let is_global = poly_trait_ref.trait_ref.path.is_global();
+            let is_local = if let Some(def_id) = poly_trait_ref.trait_ref.trait_def_id() {
```

Comment:
I don't think we need this check.

Reply:
Right! with this is local I will ban suggestions for the following code, but the blanket implementation should be fine in this case

```rust
impl fmt::Display for LocalTraitOne {
//~^ ERROR trait objects must include the `dyn` keyword
    fn fmt(&self, _f: &mut fmt::Formatter<'_>) -> fmt::Result {
        todo!();
    }
}
```

Reply:
just make sure this is what's being suggested.

NO: `impl fmt::Display for LocalTrait {}`
YES: `impl LocalTrait for fmt::Display {}` -> `impl<T: Display> LocalTrait for T {}`

Reply:
Seems to look good right

```
   = warning: this is accepted in the current edition (Rust 2015) but is a hard error in Rust 2021!
   = note: for more information, see <https://doc.rust-lang.org/nightly/edition-guide/rust-2021/warnings-promoted-to-error.html>
help: use `dyn`
   |
29 - impl LocalTraitOne for fmt::Display {}
29 + impl LocalTraitOne for dyn fmt::Display {}
   | 
help: alternatively use a blanket implementation to implement `LocalTraitOne for all types that also implement `fmt::Display`
   |
29 | impl<T: fmt::Display> LocalTraitOne for T {}
   |     +++++++++++++++++                   ~
```
and in the case that I posted, no blanked suggestion

```
warning: trait objects without an explicit `dyn` are deprecated
  --> src/test/ui/suggestions/suggest-blanket-impl-local-trait.rs:16:23
   |
16 | impl fmt::Display for LocalTraitOne {
   |                       ^^^^^^^^^^^^^
   |
   = warning: this is accepted in the current edition (Rust 2015) but is a hard error in Rust 2021!
   = note: for more information, see <https://doc.rust-lang.org/nightly/edition-guide/rust-2021/warnings-promoted-to-error.html>
help: use `dyn`
   |
16 - impl fmt::Display for LocalTraitOne {
16 + impl fmt::Display for dyn LocalTraitOne {

```

Reply:
Cool

Reply:
Thanks for suggestion

Reply:
Wait, I didn't understand this conversation. Please remove this.

The suggestion is being skipped in this case.

```
error[E0782]: trait objects must include the `dyn` keyword
  --> $DIR/suggest-blanket-impl-local-trait.rs:34:24
   |
LL | impl LocalTraitOne for fmt::Display {}
   |                        ^^^^^^^^^^^^
   |
help: add `dyn` keyword before this trait
   |
LL - impl LocalTraitOne for fmt::Display {}
LL + impl LocalTraitOne for dyn fmt::Display {}
   |
```

This should also be suggesting `impl<T: Display> LocalTraitOne for T {}`

Reply:
Damnt! I'm stupid, I did not push the change for this review, yeah in the code there is still the check regarding this review here.

Sorry!